context.go

```diff
@@ -702,11 +703,16 @@ type FileHandler func(*parser.File)
 // which the future output will depend is returned.  This list will include both
 // Blueprints file paths as well as directory paths for cases where wildcard
 // subdirs are found.
+//
+// syncVisitor will be called synchronously on each discovered file, and will only be called
```

Comment:
This seems very abstract, especially since WalkBlueprintsFiles is only used once.  Maybe instead of adding syncVisitor just so you can pass a function to count goroutines, and since WalkBlueprintsFiles is now the one starting the goroutines, WalkBlueprintsFiles should count the goroutines itself and not exit until they all return?

Reply:
sounds good